crac/src/main/java/io/micronaut/crac/resources/RefreshEventResource.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.crac.resources;
+
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.event.ApplicationEventPublisher;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.crac.CracConfigurationProperties;
+import io.micronaut.crac.OrderedResource;
+import io.micronaut.runtime.context.scope.refresh.RefreshEvent;
+import jakarta.inject.Singleton;
+import org.crac.Context;
+import org.crac.Resource;
+
+/**
+ * An OrderedResource that emits a RefreshEvent when a CRaC checkpoint is taken.
+ *
+ * @author Tim Yates
+ * @since 1.0.0
+ */
+@Singleton
+@Requires(property = CracConfigurationProperties.PREFIX + ".refresh-beans", defaultValue = StringUtils.TRUE, value = StringUtils.TRUE)
```

Comment:
should we also inject in the constructor CracEventPublisher and publish events in this resource as well?

Reply:
Good call..  Done in https://github.com/micronaut-projects/micronaut-crac/pull/18/commits/e5f0fbc45a5c2c4b3d7f86908ef95fda42122adb